lib/matplotlib/tests/test_axes.py

```diff
@@ -4603,6 +4603,36 @@ def test_hist_stacked_bar():
     ax.legend(loc='upper right', bbox_to_anchor=(1.0, 1.0), ncols=1)
 
 
+@pytest.mark.parametrize("histtype", ["step", "stepfilled"])
+@pytest.mark.parametrize("color", [["blue", "green", "brown"], [None]*3])
+@pytest.mark.parametrize("edgecolor", [["red", "black", "blue"], [None]*3])
+@pytest.mark.parametrize("facecolor", [["blue", "green", "brown"], [None]*3])
+@pytest.mark.parametrize("linewidth", [[1, 1.5, 2], [None]*3])
+@pytest.mark.parametrize("hatch", [["/", "\\", "."], [None]*3])
+@pytest.mark.parametrize("linestyle", [["-", "--", ":"], [None]*3])
+@check_figures_equal(extensions=["png"])
+def test_hist_vectorized_params(fig_test, fig_ref, histtype, color, edgecolor,
```

Comment:
Please don't use multiple parametrizations. Creating and comparing two figures is quite expensive (>200ms). You create 2^5 = 32 tests, i.e. > 6s test execution time. Note that we run thousands of tests and we cannot afford inefficient tests.

The parameters do not depend on each other, so testing the product space is not necessary. You can either write separate tests per parameter, or use one `parametrize` decorator with mutliple variables.

Reply:
As the color, edgecolor and facecolor parameters depend on each other, I can leave them as it is and test the rest of them only for the set case, so that we don't miss cases mentioned in the table. Or should we just test for all of the variables set individually nevertheless?